elastica/rod/factory_function.py

```diff
@@ -323,6 +323,324 @@ def allocate(
     )
 
 
+def allocate_ring_rod(
+    n_elements,
+    ring_center_position,
+    direction,
+    normal,
+    base_length,
+    base_radius,
+    density,
+    nu,
+    youngs_modulus: float,
+    nu_for_torques: Optional[float] = None,
+    shear_modulus: Optional[float] = None,
+    position: Optional[np.ndarray] = None,
+    directors: Optional[np.ndarray] = None,
+    rest_sigma: Optional[np.ndarray] = None,
+    rest_kappa: Optional[np.ndarray] = None,
+    *args,
+    **kwargs,
+):
```

Comment:
this function has a lot of common steps with a straight rod? maybe some common features can be factored out in another function?

Reply:
Lets do this in another PR. I created and issue for this #240